### ISSUE TITLE
[New Resource ]: Assign accounts to OU (aws_organizations_ou_assignment)

### ISSUE THREAD

**Issue by rnikoopour** [hashicorp/terraform-provider-aws#30658]
### Description

There is no way to assign an existing AWS account to an OU within an org.  Currently the only way to do this is via the `parent_id` field on the `aws_organizations_account` resource.

A new resource `aws_organizations_ou_assignment` could enable folks to assign accounts to an OU without having to have created it via `aws_organizations_account` resource.

This would used in conjunction with the data source requested in #30656 and #30675 (or via hardcoded values)

### Requested Resource(s) and/or Data Source(s)

* aws_organizations_ou_assignmnet

### Potential Terraform Configuration

```terraform
data "aws_organizations_account" "my_account" {
    id = "1234567890"
}

data "aws_organizations_ou" "my_ou" {
    parent_id = "ou-1234567890"
    name = "my_ou"
}

resource "aws_orgnizations_ou_assignment" "my_ou_assignment" {
   ou_id = data.aws_organizations_ou.my_ou.id
   assignments = [data.aws_organizations_account.my_account.id]
}
```

### Would you like to implement a fix?

None

**Comment by dcibarra** [comment]
Yes please! Frustrating to have to do this manually. Should be possible since there's an API endpoint
https://docs.aws.amazon.com/organizations/latest/APIReference/API_MoveAccount.html

**Comment by gdavison** [comment]
Hi @rnikoopour and @dcibarra. To clarify, are you asking to move an Organizations Account that isn't managed by Terraform into a different OU? If the Account is managed by Terraform, updating `parent_id` will move the Account into that OU. Note that it is possible to import an existing Account into Terraform management.

**Comment by rnikoopour** [comment]
That is correct.  I don't create accounts via TF, but I'd to manage the structure of my organization via TF.  And I currently am unable to do so since I cannot assign accounts to OUs that weren't created via Terraform.
